pkg/vere/io/http.c

```diff
@@ -2695,14 +2816,79 @@ _http_stream_slog(void* vop_p, c3_w pri_w, u3_noun tan)
   u3z(tan);
 }
 
+/* _http_spin_timer_cb(): send heartbeat to slog streams and restart timer
+*/
+static void
+_http_spin_timer_cb(uv_timer_t* tim_u)
+{ 
+  u3_httd* htd_u = tim_u->data;
+  u3_hreq* siq_u = htd_u->fig_u.siq_u;
+
+  if ( 0 != siq_u ) {
+    c3_w siz_w      = 1024;
+    c3_c* buf_c     = c3_malloc(siz_w);
+    u3t_spin* stk_u = htd_u->stk_u;
+    c3_w pos_w      = stk_u->off_w;
+    c3_w out_w      = 0;
+
+    while (pos_w > 4) {
+      c3_w  len_w;
+      pos_w -=4;
+
+      if ( siz_w < out_w + 4 ) {
+         buf_c = c3_realloc(buf_c, siz_w*2);
+         siz_w *= 2;
+      }
+
+      memcpy(&len_w, &stk_u->dat_y[pos_w], 4);
+      pos_w -= len_w;
+
+      if ( siz_w < out_w + 4 ) {
+         buf_c = c3_realloc(buf_c, siz_w*2);
+      }
+      memcpy(buf_c + out_w, "/", 1);
+      out_w +=1;
+
+      if ( siz_w < out_w + len_w ) {
+         buf_c = c3_realloc(buf_c, siz_w*2);
+      }
+
+      memcpy(buf_c + out_w, &stk_u->dat_y[pos_w], len_w);
+      out_w += len_w;
+      buf_c[out_w] = '\0';
+    }
+    buf_c[out_w] = '\0';
+
+    if ( 0 != stk_u->off_w ) {
+      u3_noun tan = u3i_string(buf_c);
+      u3_noun lin = u3i_list(u3i_string("data:"),
+                             tan,
+                             c3_s2('\n', '\n'),
+                             u3_none);
+      u3_atom txt = u3qc_rap(3, lin);
+      u3_noun dat = u3nt(u3_nul, u3r_met(3, txt), txt);
+
+      while ( 0 != siq_u ) {
+        _http_continue_respond(siq_u, u3k(dat), c3n);
+        siq_u = siq_u->nex_u;
+      }
```

Comment:
Need a `u3z(dat); u3z(lin);` after the loop to avoid leaking them. (You could also use or add `u3kc_rap()` and just deal with `dat`.

Reply:
Done